src/protocol/sort/shuffle.rs

```diff
@@ -0,0 +1,99 @@
+use crate::helpers::prss::PrssSpace;
+use rand::rngs::mock::StepRng;
+use rand::seq::SliceRandom;
+
+#[allow(dead_code)]
+pub struct Shuffle {}
+
+impl Shuffle {
+    /// The Fisher–Yates shuffle is an algorithm for generating a random permutation of a finite
+    /// sequence—in plain terms, the algorithm shuffles the sequence. The algorithm effectively
+    /// puts all the elements into a hat; it continually determines the next element by randomly
+    /// drawing an element from the hat until no elements remain. The algorithm produces an unbiased
+    /// permutation: every permutation is equally likely. This algorithm takes time proportional to
+    /// the number of items being shuffled and shuffles them in place.
```

Comment:
I'd recommend linking to the Wikipedia page from which you're taking this text: https://en.wikipedia.org/wiki/Fisher%E2%80%93Yates_shuffle